dkulib/parallelizer/parallelizer.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+"""Applies a function to a pandas DataFrame with parallelization, error logging and progress tracking"""
+
+import logging
+import inspect
+import math
+
+from collections import namedtuple
+from collections import OrderedDict
+from concurrent.futures import as_completed
+from concurrent.futures import ThreadPoolExecutor
+from copy import deepcopy
+from enum import Enum
+from time import perf_counter
+from typing import Any
+from typing import AnyStr
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+from more_itertools import chunked
+from more_itertools import flatten
+import pandas as pd
+from tqdm.auto import tqdm as tqdm_auto
+
+from dkulib.io_utils.plugin_io_utils import generate_unique
+
+
+class ErrorHandling(Enum):
+    """Enum class to identify how to handle API errors"""
+    LOG = "Log"
+    FAIL = "Fail"
+
+
+class BatchError(ValueError):
+    """Custom exception raised if the Batch function fails"""
+
+
+def _parse_batch_response_default(batch: List[Dict],
+                                  response: Any,
+                                  output_column_names: NamedTuple) -> List[Dict]:
+    """Adds the response column to the row dictionary at batch[0], while keeping the 
+    existing dict entries. Should only be used when batch_size=1.
```

Comment:
Why can't is be used for batch processing ? I think it is easy to make it work for batch_size > 1

Reply:
Yeah good point - I can open a PR for this if okay with @alexcombessie ?

Reply:
I think it can be done here. Do you think it will be long to change ?

Reply:
Made it a PR; lmk if it's as you expected
https://github.com/dataiku/dss-plugin-dkulib/pull/18

Reply:
I think we need to update the above comment as well

Reply:
What do you mean?

Reply:
I think this function can also be used for batches so I don't understand why it is written `Should only be used when batch_size=1.`

Reply:
Ah yes good point, the comment is not appropriate. Updated in 2a197b8, let me know if it is clearer.

Reply:
lgtm